python/hail/api2/matrixtable.py

```diff
@@ -2091,6 +2091,33 @@ def repartition(self, num_partitions, shuffle=True):
         jvds = self._jvds.coalesce(num_partitions, shuffle)
         return MatrixTable(jvds)
 
+    @handle_py4j
+    @typecheck_method(max_partitions=integral)
+    def naive_coalesce(self, max_partitions):
+        """Naively descrease the number of partitions.
+
+        Example
+        -------
+        Naively repartition to 10 partitions:
+
+        >>> dataset_result = dataset.naive_coalesce(10)
+
+        Warning
+        -------
+        :meth:`naive_coalesce` simply combines adjacent partitions to achieve the desired number.  It does not attempt to rebalance, unlike :meth:`repartition`, so it can produce a heavily unbalanced dataset.  An unbalanced dataset can be inefficient to operate on because the work is not evenly distributed across partitions.
```

Comment:
Line too long. Use this tool to format (set spaces to 8, paste, hit enter)

https://hail.is/tools/formatter.html